packages/core/contracts/proxy-scripts/lsp-broker/LspUniswapV2Broker.sol

```diff
@@ -15,27 +18,197 @@ import "../../financial-templates/long-short-pair/LongShortPair.sol";
  * @notice Helper contract to facilitate batched LSP and UniswapV2 transactions, including Mint+Sell and Mint+LP.
  */
 contract LspUniswapV2Broker {
+    using FixedPoint for FixedPoint.Signed;
     using FixedPoint for FixedPoint.Unsigned;
 
     /**
      * @notice Mint long and short tokens and deposit them all into a UniV2 Pool.
+     * @dev The caller of this method needs to approve `amountCollateral` collateral to be spent by this contract.
+     * @param callingAsEOA If True, caller has balance of collateral and expects to receive back all long/short tokens.
+     * @param longShortPair LSP contract address to mint position on.
+     * @param router Contract to call to exchange long and short tokens.
+     * @param amountCollateral Amount of collateral to deposit and mint long and short tokens against.
      */
-    function atomicMintAddLiquidity() public {
-        /* TODO */
+    function atomicMintAddLiquidity(
+        bool callingAsEOA,
+        LongShortPair longShortPair,
+        IUniswapV2Router01 router,
+        uint256 amountCollateral,
+        uint256 deadline
+    ) public {
+        require(address(longShortPair) != address(0), "Invalid long short pair");
+        require(address(router) != address(0), "Invalid router");
+        require(amountCollateral != 0, "Collateral to mint with");
+
+        IERC20 collateralToken = IERC20(longShortPair.collateralToken());
+        IERC20 longToken = IERC20(longShortPair.longToken());
+        IERC20 shortToken = IERC20(longShortPair.shortToken());
+
+        // 0) Pull collateral from caller if necessary and approve LSP to spend it.
+        if (callingAsEOA)
+            TransferHelper.safeTransferFrom(address(collateralToken), msg.sender, address(this), amountCollateral);
+
```

Comment:
```suggestion
```

Reply:
I like grouping together actions that fall under the same comment. So the `transferFrom` and the `approve` both execute the comment: `0) Pull collateral from caller if necessary and approve LSP to spend it.`